src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs

```diff
@@ -665,16 +651,14 @@ private void OnHostingHttpRequestInStart(object arg)
 
         private void RaiseIntrumentationEvent(IDatadogSecurity security, HttpContext context, HttpRequest request)
         {
-            string url = GetUrl(request);
-
-            var dict = new Dictionary<string, object>()
-            {
-                { "server.request.method", request.Method },
-                { "server.request.uri.raw", url },
-                { "server.request.query", request.QueryString.ToString() },
-            };
+            var dic = request.PrepareArgsForWaf();
+            security.InstrumentationGateway.RaiseEvent(dic, new HttpTransport(context));
+        }
 
-            security.InstrumentationGateway.RaiseEvent(dict, new HttpTransport(context));
+        private void RaiseIntrumentationEvent(IDatadogSecurity security, HttpContext context, HttpRequest request, RouteData routeDatas)
```

Comment:
I think you could merge the two `RaiseIntrumentationEvent` by make `routeDatas` optional.

('Data' is uncountable in English, so always singular)